### PR TITLE
Tests for api contracts

### PR DESCRIPTION
**Pull request by vikhyat187**
Added integration tests method declaration.

- [ ] Adding the dummy data is pending, will once explore that and add to it, the fixtues

Test status 0 0 0 

API contracts - #8 

Closes ticket #10

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]
please add test case stats

**Review by iamitprakash — DISMISSED** [review]

File: skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+                .build());
+
+        Response response = given()
+                .contentType("application/json")
+                .body(requestBody)
+                .post("/endorsements");
+        response.then()
+                .statusCode(200)
+                .body("code", equalTo(200))
+                .body("message", equalTo("Created endorsement sucessfully"));
```
Comment (heyrandhir) [inline skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java:125]: Nit: its generally a good idea to return the newly created resource details in the POST Response as well.

File: skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    public void testAPIReturns200_OnEndorsementsUpdation(){
+        String endorsementId = "e-1";
+        Response response = given()
+                .pathParam("endorsementId", endorsementId)
+                .patch("/endorsements");
+
+        response.then()
+                .statusCode(200)
+                .body("code",equalTo(200))
+                .body("message", equalTo("Updated endorsement sucessfully"));
```
Comment (heyrandhir) [inline skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java:142]: Nit: its generally a good idea to return the updated resource details in the PATCH Response as well.

File: skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  "endorsementType": "POSITIVE",
+  "endorsersList": [
+    {
+      "endorserId": "user-2",
+      "description": "string",
+      "userType": "NORMAL_USER"
+    }
+  ]
+}
+         */
```
Comment (heyrandhir) [inline skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java:183]: Is it for reference purpose ? or by mistake missed to remove. if its for reference can we format it please.

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+      "endorsersList": [
+        {
+          "endorserId": "user-2",
+          "description": "string",
+          "userType": "NORMAL_USER"
+        }
+      ]
+    }
+  ]
+}        */
```
[the remark this answers is not in the document]
Comment (vikhyat187) [reply skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java:48]: Have added just for reference

File: skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java (hunk 3 above)
Comment (vikhyat187) [reply skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java:183]: Its for reference, will get it formatted

File: skill-tree /src/main/java/com/RDS/skilltree/response/EndorsementResponse.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.RDS.skilltree.response;
+
+import com.RDS.skilltree.enums.EndorsementType;
+import com.RDS.skilltree.enums.Status;
+
+import java.util.List;
+
+public class EndorsementResponse {
```
[the remark this answers is not in the document]
Comment (vikhyat187) [reply skill-tree /src/main/java/com/RDS/skilltree/response/EndorsementResponse.java:8]: Added it, it was missed by me

File: skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        Response response = given()
+                .get("/endorsements?status=PENDING");
+
+        response.then()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("data", hasSize(1))
+                .body("data[0].endorseeId",equalTo("user-1")) //TODO update this
+                .body("data[0].skillName",equalTo("Java"))
+                .body("data[0].status", anyOf(equalTo("APPROVED"), equalTo("REJECTED"), equalTo("PENDING")))
```
[the remark this answers is not in the document]
Comment (vikhyat187) [reply skill-tree /src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java:82]: correct

**Comment by vikhyat187** [comment]
Hey @heyrandhir @iamitprakash can you please approve the #14 for changing the API contracts, as per the changes in the response object

**Review by heyrandhir — DISMISSED** [review]

File: skill-tree /src/main/java/com/RDS/skilltree/enums/Behaviour.java [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package com.RDS.skilltree.enums;
+
+public enum Behaviour {
```
Comment (iamitprakash) [inline skill-tree /src/main/java/com/RDS/skilltree/enums/Behaviour.java:3]: can't we have common enum for Behaviour and EndorsementType

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: skill-tree /src/main/java/com/RDS/skilltree/enums/Behaviour.java (hunk 7 above)
Comment (vikhyat187) [reply skill-tree /src/main/java/com/RDS/skilltree/enums/Behaviour.java:3]: Yes we can, will make this change

**Review by heyrandhir — APPROVED** [review]

**Pull request merged**